agent/wsclient/client.go

```diff
@@ -113,6 +118,8 @@ type ClientServerImpl struct {
 	AnyRequestHandler RequestHandler
 	// URL is the full url to the backend, including path, querystring, and so on.
 	URL string
+	// Lock needed to ensure that only one routine is writing to the socket
```

Comment:
nit: Although there isn't a lint format defined for package scoped var's, i'd prefer it if they followed the style for global var's. For example, this comment would be `writeLock is needed to ...`. Basically, the comment starts with the name of the variable of the method name. https://golang.org/doc/effective_go.html#commentary